### PR TITLE
providers/ec2: update metadata version

### PR DESCRIPTION
**Pull request by margamanterola**
# Update the metadata version in our ignition fork

The current date is so old that it leads to errors in some cases.

See https://github.com/kinvolk/Flatcar/issues/220 and https://github.com/coreos/ignition/issues/989

# Testing done

I haven't really tested this and I'm not sure how to :-/

### CODE REVIEW

File: internal/providers/ec2/ec2.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ var (
 	userdataUrl = url.URL{
 		Scheme: "http",
 		Host:   "169.254.169.254",
-		Path:   "2009-04-04/user-data",
+		Path:   "2019-10-01/user-data",
```
Comment (invidian) [inline internal/providers/ec2/ec2.go:39]: If the format of obtained data exactly the same?

Comment (margamanterola) [reply internal/providers/ec2/ec2.go:39]: I don't know, I'm just stumbling in the dark here. But from reading https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html, it looks like new versions just add new fields, so the format should stay the same, it just has additional fields.

Upstream ignition changed the date without adding any other changes, as far as I could see.

Comment (hakman) [reply internal/providers/ec2/ec2.go:39]: I am wondering why this approach was used 5 years ago and if not outdated.
Even in the same function there is a call to the `ec2metadata` API which can also be used to retrieve the userdata also:
https://docs.aws.amazon.com/sdk-for-go/api/aws/ec2metadata/#EC2Metadata.GetUserData
```
	sess := session.Must(session.NewSession())
	svc := ec2metadata.New(sess)
	userdata, err := svc.GetUserData()
```

**Comment by pothos** [comment]
I think we can close this PR and if needed, create an upstream PR

**Pull request closed without merging**